File: frontend/src/types/production/notification.ts
// 生产模块通知配置 TypeScript 类型

export interface ProductionNotificationConfig {
  notify_type: string
  name: string
  description: string
  is_enabled: boolean
  extra_user_ids: string[]
}

export interface NotificationConfigUpdateInput {
  is_enabled: boolean
  extra_user_ids: string[]
}

File: backend/app/modules/production/schemas/notification.py
"""通知配置 API 契约。"""

import uuid

from pydantic import BaseModel, Field


class NotificationConfigOut(BaseModel):
    """单个通知类型的配置（无配置行时按默认值返回）。"""

    notify_type: str
    name: str
    description: str
    is_enabled: bool
    extra_user_ids: list[uuid.UUID]


class NotificationConfigUpdateIn(BaseModel):
    """更新通知配置入参。"""

    is_enabled: bool = Field(description="是否启用该通知")
    extra_user_ids: list[uuid.UUID] = Field(
        default_factory=list, description="额外通知人员（发送时与功能接收人合并去重）",
    )
